cmd/_integration-tests/transport/http_test.go

```diff
@@ -400,6 +404,124 @@ func TestStrangeRPCName(t *testing.T) {
 	}
 }
 
+// Test that if a truss client receives a non-json response from a "truss"
+// server, that we put that response body in the error message. To allow for
+// developers to see the request body in the errors.
+func TestNonJSONResponseBodyFromClientCallIsInError(t *testing.T) {
+	svchttp, err := httpclient.New(nonJSONHTTPAddr)
+	if err != nil {
+		t.Fatalf("cannot create httpclient: %q", err)
+	}
+
+	var req pb.Empty
+	_, err = svchttp.ErrorRPCNonJSON(context.Background(), &req)
+	if err == nil {
+		t.Fatal("Expected error from non-json response with http client")
+	}
+
+	if !strings.Contains(err.Error(), brokenHTTPResponse) {
+		t.Fatalf("Expected error to contain `%s`; error is `%s`", brokenHTTPResponse, err.Error())
+	}
+}
+
+// Test that if a non json response is recieved that is greater than 8KB, that
+// we only put the first 8KB error, as to not flood memory with huge errors.
+func TestNonJSONResponseBodyFromClientCallIsLessThan8KB(t *testing.T) {
+	svchttp, err := httpclient.New(nonJSONHTTPAddr)
+	if err != nil {
+		t.Fatalf("cannot create httpclient: %q", err)
+	}
+
+	var req pb.Empty
+	_, err = svchttp.ErrorRPCNonJSONLong(context.Background(), &req)
+	if err == nil {
+		t.Fatal("Expected error from non-json response with http client")
+	}
+
+	l := len(err.Error())
+	// Add 200 for padding for the actual error message in addition to the response body
+	if l > 8196+200 {
+		t.Fatalf("Expected error to be less than 8KB with a little padding, actual %d", l)
+	}
+	t.Log("Non JSON response length", l)
+}
+
+// Test that if a truss server receives a non-json request, we put that request
+// body in the error message. To allow for developers to see the request body in the errors.
+func TestNonJSONRequestBodyIsInError(t *testing.T) {
+	// Put some bad data into the body
+	req, err := http.NewRequest("GET", httpAddr+"/error/non/json", strings.NewReader(brokenHTTPRequest))
```

Comment:
The semantics for GET requests are pretty explicitly spelled out in [the RFC for HTTP 1.1 as being intended to retrieve data specified in the URI of a request](https://tools.ietf.org/html/rfc2616#section-9.3). So while you *can* put a body into a GET request, it's unusual to do so. It makes much more sense for these to be POST requests and for the associated RPCs in the `.proto` file to serve data from a `post` as well.